l1-contracts/src/core/libraries/crypto/SampleLib.sol

```diff
@@ -32,203 +41,67 @@ library SampleLib {
-   *          To be more clever, we can compute the `shuffeRounds` and `pivots` separately
-   *          such that they get shared accross multiple indices.
-   *
-   * @param _committeeSize - The size of the committee
-   * @param _indexCount - The total number of indices
-   * @param _seed - The seed to use for shuffling
-   *
-   * @return indices - The indices of the committee
-   */
-  function computeCommitteeClever(uint256 _committeeSize, uint256 _indexCount, uint256 _seed)
-    internal
-    pure
-    returns (uint256[] memory)
-  {
-    uint256[] memory indices = new uint256[](_committeeSize);
+    uint256 upperLimit = _indexCount - 1;
 
     for (uint256 index = 0; index < _committeeSize; index++) {
-      uint256 originalIndex = computeOriginalIndex(index, _indexCount, _seed);
-      indices[index] = originalIndex;
-    }
+      uint256 sampledIndex = computeShuffledIndex(index, upperLimit, _seed);
 
-    return indices;
-  }
+      // Get index, or its swapped override
+      sampledIndices[index] = getValue(sampledIndex);
 
-  /**
-   * @notice  Computes the shuffled index
-   *
-   * @param _index - The index to shuffle
-   * @param _indexCount - The total number of indices
-   * @param _seed - The seed to use for shuffling
-   */
-  function computeShuffledIndex(uint256 _index, uint256 _indexCount, uint256 _seed)
-    internal
-    pure
-    returns (uint256)
-  {
-    require(_index < _indexCount, Errors.SampleLib__IndexOutOfBounds(_index, _indexCount));
-    uint256 rounds = computeShuffleRounds(_indexCount);
-
-    uint256 index = _index;
+      // Swap with the last index
+      setOverrideValue(sampledIndex, getValue(upperLimit));
 
-    for (uint256 currentRound = 0; currentRound < rounds; currentRound++) {
-      uint256 pivot = computePivot(_seed, currentRound, _indexCount);
-      index = computeInner(_seed, pivot, index, currentRound, _indexCount);
+      // Decrement the upper limit
+      upperLimit--;
     }
 
-    return index;
-  }
 
-  /**
-   * @notice  Computes the original index from a shuffled index
-   *
-   * @dev     Notice, that we are using same logic as `computeShuffledIndex` just looping in reverse.
-   *
-   * @param _shuffledIndex - The shuffled index
-   * @param _indexCount - The total number of indices
-   * @param _seed - The seed to use for shuffling
-   *
-   * @return index - The original index
-   */
-  function computeOriginalIndex(uint256 _shuffledIndex, uint256 _indexCount, uint256 _seed)
-    internal
-    pure
-    returns (uint256)
-  {
-    require(
-      _shuffledIndex < _indexCount, Errors.SampleLib__IndexOutOfBounds(_shuffledIndex, _indexCount)
-    );
+    return sampledIndices;
+  }
 
-    uint256 rounds = computeShuffleRounds(_indexCount);
+  function getValue(uint256 _index) internal view returns (uint256) {
+    uint256 overrideValue = getOverrideValue(_index);
+    if (overrideValue != 0) {
+      // TODO: edge case where override actually is 0
```

Comment:
I think the only case we could end up there would be if index have not been zero yet and you hit `upper_limit = 0`, but in that case both values values be 0 so it don't matter. And you would also not be entering the loop again and actually reading that value 👀